src/components/Buttons/__tests__/Checkbox.spec.js

```diff
@@ -27,7 +27,7 @@ describe('Checkbox', () => {
     })
   })
 
-  it('sets isSelected to false if isDisabled is set to true', () => {
+  it('disables the input when it is disabled', () => {
```

Comment:
shouldn't do test refactoring at the same time I am changing code, but just for future reference:

How do you feel about these tests ... I am deeply uncomfortable with this. A lot of our snacks tests seem to be testing components through inspecting internal implementation by either examining properties of subcomponents or inspecting values of internal state and/or props. 

Is this the lay of the land in the react community nowadays? It feels super brittle to me, shouldn't tests be concerned with outside effects (such as colors of the component and whether or not it responds to click events in this example) and not in its internal guts.

Reply:
In general I'm 100% with you, I agree with avoiding testing internals (so I'd rarely/never set a prop directly or call setState from within a test). The only way we should be interacting with components in test is how a user would (e.g. click/change events).

I'm unclear on what you mean exactly with "examining  properties of subcomponents" though, I generally don't have a problem with assertions like `expect(htmlBtn.checked).toBe(true)` since it's checking actual DOM output and not component internals.

Reply:
yeah maybe you are right, I was scared by some tests that where accessing states before.

Just feel like wrapper.find('input').instance() is a bit problematic because it relies on the input element actually being rendered inside a component ... but maybe I am paranoid.

Reply:
example form tooltip spec:
expect(tooltip.props().isVisible).toEqual(true)
    expect(tooltip.state().show).toEqual(false)

Reply:
Yeah, agreed on the last two examples. The tests accomplish what they want but rely too much on implementation details. but it's similar to testing private methods in ruby, best to avoid it and if something is complex enough it can be extracted out and tested on its own. `wrapper.find('input').instance()` feels ok because it's finding the rendered output, I'm def open to arguments either way though.

Reply:
yeah fair enough, I think thats reasonable ... the find('input') is no different from snapshot test in a way.

lets just not do props and state inspection if we can avoid it :)